query_server/query/src/metadata.rs

```diff
@@ -166,6 +176,183 @@ impl MetaData for LocalCatalogMeta {
+            database_name = self.database_name.as_str();
+        }
+
+        match self
+            .engine
+            .get_table_schema(database_name, table_name)
+            .unwrap()
+        {
+            None => Err(MetadataError::TableNotExists {
+                table_name: table.to_string(),
+            }),
+            Some(table_schema) => {
+                let schema = Arc::new(Schema::new(vec![
+                    Field::new("fieldname", DataType::Utf8, false),
+                    Field::new("type", DataType::Utf8, false),
+                    Field::new("istag", DataType::Boolean, false),
+                    Field::new("compression", DataType::Utf8, false),
+                ]));
+                // fieldname    type        istag       compression
+                //      time    Time,       No          codec
+                //      c1      String      No          codec
+                //      c2      uint64      No          default
+                let columns = table_schema.columns().clone();
+
+                let mut name_column = vec![];
+                let mut type_column = vec![];
+                let mut tags = vec![];
+                let mut compressions = vec![];
+
+                for item in &columns {
+                    let field_name = item.name.as_str();
+                    let field_type;
+                    let mut tag = false;
+
+                    match item.column_type {
+                        ColumnType::Tag => {
+                            field_type = "STRING";
+                            tag = true;
+                        }
+                        ColumnType::Time => field_type = "TIMESTAMP",
+                        ColumnType::Field(ValueType::Float) => field_type = "DOUBLE",
+                        ColumnType::Field(ValueType::Integer) => field_type = "BIGINT",
+                        ColumnType::Field(ValueType::Unsigned) => field_type = "UNSIGNED",
+                        ColumnType::Field(ValueType::String) => field_type = "STRING",
+                        ColumnType::Field(ValueType::Boolean) => field_type = "BOOLEAN",
+                        ColumnType::Field(ValueType::Unknown) => field_type = "UNKNOW",
+                    }
+
+                    let compression = match Encoding::from(item.codec) {
+                        Encoding::Default => "Default",
+                        Encoding::Null => "Null",
+                        Encoding::Delta => "Delta",
+                        Encoding::Quantile => "Quantile",
+                        Encoding::Gzip => "Gzip",
+                        Encoding::Bzip => "Bzip",
+                        Encoding::Gorilla => "Gorilla",
+                        Encoding::Snappy => "Snappy",
+                        Encoding::Zstd => "Zstd",
+                        Encoding::Zlib => "Zlib",
+                        Encoding::BitPack => "BitPack",
+                        Encoding::Unknown => "Unknown",
```

Comment:
Would it be better to extract method of Encoding

Reply:
good point